lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1497,6 +1563,43 @@ Value FIRRTLLowering::getLoweredAndExtendedValue(Value value, Type destType) {
     return getOrCreateIntConstant(destWidth, 0);
   }
 
+  if (auto array = result.getType().dyn_cast<hw::ArrayType>()) {
+    // srcWidth == unsigned(destWidth) --> ok
+    // srcWidth > unsigned(destWidth)  --> error
+    // srcWidth < unsigned(destWidth)  --> insert concat/sext to elements
+
+    auto baseDestType = destType.cast<FIRRTLType>().getVectorBaseType();
+    auto baseSourceType =
+        value.getType().cast<FIRRTLType>().getVectorBaseType();
+    auto baseResultType = getVectorBaseType(result.getType());
+    if (!baseSourceType || !baseDestType || !baseResultType)
+      return {};
+
+    auto srcWidth = baseResultType.cast<IntegerType>().getWidth();
+    auto destWidth = baseDestType.cast<IntType>().getWidthOrSentinel();
+    if (destWidth == -1)
+      return {};
+
+    if (srcWidth == unsigned(destWidth))
+      return result;
```

Comment:
This handles the case where a vector value is used as a type with different element widths, but does firrtl also allow  changing the vector length?  E.g. can you connect things like a `<3x i32>` and `<2x i32>`?

Reply:
Thank you for pointing it out. Yes, firrtl allow changing the vector length. I will modify them.